channeldb/invoices.go

```diff
@@ -519,9 +526,16 @@ func (d *DB) AddInvoice(newInvoice *Invoice, paymentHash lntypes.Hash) (
 			return ErrDuplicateInvoice
 		}
 
+		// Check that we aren't inserting an invoice with a duplicate
+		// payment address. The all-zeros payment address is
+		// special-cased to support legacy keysend invoices which don't
+		// assign one. This is safe since later we also will avoid
+		// indexing them and avoid collisions.
 		payAddrIndex := tx.ReadWriteBucket(payAddrIndexBucket)
-		if payAddrIndex.Get(newInvoice.Terms.PaymentAddr[:]) != nil {
-			return ErrDuplicatePayAddr
+		if newInvoice.Terms.PaymentAddr != LegacyKeysendPayAddr {
```

Comment:
Is it right that we only execute this check and the one below to help users that have been running master? I think a migration that deletes the all-zeroes address from the index is worth considering.

Reply:
imo it's pretty unlikely that this is affecting anyone in the wild, there isn't necessarily any harm in leaving it for now. that said we should definitely consider it if we hear reports or it conflicts with things we want to do in the future

Reply:
So no migration that is? Then I'd say at least add a clear comment that this is just needed because of a bug that has lived in master for a while.

Reply:
not sure i follow, this check would have been required whether or not the bug was in master or not. what would you like me to add?